solidity/contracts/bridge/Fraud.sol

```diff
@@ -111,20 +95,22 @@ library Frauds {
     ///        for reference
     /// @param signature Bitcoin signature in the R/S/V format
     /// @dev Requirements:
+    ///      - Wallet behind `walletPublicKey` must be in `Live` or `MovingFunds`
+    ///        state
     ///      - The challenger must send appropriate amount of ETH used as
     ///        fraud challenge deposit
     ///      - The signature (represented by r, s and v) must be generated by
-    ///        the wallet behind `walletPubKey` during signing of `sighash`
+    ///        the wallet behind `walletPublicKey` during signing of `sighash`
     ///      - Wallet can be challenged for the given signature only once
-    function submitChallenge(
-        Data storage self,
+    function submitFraudChallenge(
+        BridgeState.Storage storage self,
+        Wallets.Data storage wallets,
```

Comment:
We added a new parameter 
```
Wallets.Data storage wallets
```
so we should probably add the corresponding `@param` to the function description.

Reply:
The convention so far is that we are not adding those storage pointers to the parameter list. IMO they do not require explanation and ideally, they would be passed implicitly.

Reply:
OK